py/src/nupic/algorithms/anomaly_likelihood.py

```diff
@@ -0,0 +1,680 @@
+
+
+Low-Level Function Usage
+++++++++++++++++++++++++
+
+There are two primary interface routines.
+
+- :func:`~.anomaly_likelihood.estimateAnomalyLikelihoods`: batch routine, called
+  initially and once in a while
+- :func:`~.anomaly_likelihood.updateAnomalyLikelihoods`: online routine, called
+  for every new data point
+
+Initially:
+
+.. code-block:: python
+
+   likelihoods, avgRecordList, estimatorParams = \\
+     estimateAnomalyLikelihoods(metric_data)
+
+Whenever you get new data:
+
+.. code-block:: python
+
+   likelihoods, avgRecordList, estimatorParams = \\
+     updateAnomalyLikelihoods(data2, estimatorParams)
+
+And again (make sure you use the new estimatorParams returned in the above call
+to updateAnomalyLikelihoods!).
+
+.. code-block:: python
+
+   likelihoods, avgRecordList, estimatorParams = \\
+     updateAnomalyLikelihoods(data3, estimatorParams)
+
+Every once in a while update estimator with a lot of recent data.
+
+.. code-block:: python
+
+   likelihoods, avgRecordList, estimatorParams = \\
+     estimateAnomalyLikelihoods(lots_of_metric_data)
+
+
+PARAMS
+++++++
+
+The parameters dict returned by the above functions has the following
+structure. Note: the client does not need to know the details of this.
+
+::
+
+ {
+   "distribution":               # describes the distribution
+     {
+       "name": STRING,           # name of the distribution, such as 'normal'
+       "mean": SCALAR,           # mean of the distribution
+       "variance": SCALAR,       # variance of the distribution
+
+       # There may also be some keys that are specific to the distribution
+     },
+
+   "historicalLikelihoods": []   # Contains the last windowSize likelihood
+                                 # values returned
+
+   "movingAverage":              # stuff needed to compute a rolling average
+                                 # of the anomaly scores
+     {
+       "windowSize": SCALAR,     # the size of the averaging window
+       "historicalValues": [],   # list with the last windowSize anomaly
+                                 # scores
+       "total": SCALAR,          # the total of the values in historicalValues
+     },
+
+ }
+
+"""
+
+import collections
+import math
+import numbers
+import numpy
+
+from nupic.utils import MovingAverage
+
+class AnomalyLikelihood:
+  """
+  Helper class for running anomaly likelihood computation. To use it simply
+  create an instance and then feed it successive anomaly scores:
+
+  .. code-block:: python
+
+      anomalyLikelihood = AnomalyLikelihood()
+      while still_have_data:
+        # Get anomaly score from model
+
+        # Compute probability that an anomaly has ocurred
+        anomalyProbability = anomalyLikelihood.anomalyProbability(
+            value, anomalyScore, timestamp)
+
+  """
+
+
+  def __init__(self,
+               claLearningPeriod=None,
+               learningPeriod=288,
+               estimationSamples=100,
```

Comment:
Arguments learningPeriod & estimationSamples do very similar things, can they be merged?